integration/utils/follower.go

```diff
@@ -128,9 +128,20 @@ Loop:
 		getBlockByHeightDuration := time.Since(GetBlockByHeightTime)
 
 		var blockTxs, blockUnknownTxs uint64
+
+		for _, guaranteed := range block.CollectionGuarantees {
+			_, err := f.client.GetCollection(f.ctx, guaranteed.CollectionID)
+			if err != nil {
+				continue Loop // silently try again if not ready; can happen several dozen times
+			}
+		}
+
 		for _, guaranteed := range block.CollectionGuarantees {
 			col, err := f.client.GetCollection(f.ctx, guaranteed.CollectionID)
```

Comment:
Instead of iterating over block.CollectionGuarantees and calling getCollection again here, we can just collect the results, in say `map[collectionId]*Collection`, in the above loop

Reply:
Thanks for the comment! For me, this suggestions seems to make the code unnecessarily more complicated? I have no problem to scrap this PR and feel free to re-write the code as you see fit. If anything I was hoping somebody might have a suggestion to simplify the entire loop so that my hacky fix wouldn't be necessary and the `continue Loop`s would also not be necessary :-)

Reply:
we could consider just looping here instead:

```
	col, err := f.client.GetCollection(f.ctx, guaranteed.CollectionID)
	for ; err != nil; col, err = f.client.GetCollection(f.ctx, guaranteed.CollectionID) {
		time.Sleep(time.Millisecond)
	}
```

don't know if you have to sleep in the loop, but we'd basically assume we'd always get to a point where we stop erroring. Might have to add a stop loop break after certain number of tries.